metadata-io/src/main/java/com/linkedin/metadata/graph/neo4j/Neo4jGraphService.java

```diff
@@ -90,39 +100,105 @@ public void addEdge(@Nonnull final Edge edge) {
 
     // Add/Update relationship
     final String mergeRelationshipTemplate =
-        "MATCH (source:%s {urn: $sourceUrn}),(destination:%s {urn: $destinationUrn}) MERGE (source)-[r:%s]->(destination) SET r = $properties";
-    final String statement =
-        String.format(mergeRelationshipTemplate, sourceType, destinationType, edge.getRelationshipType());
-
-    final Map<String, Object> paramsMerge = new HashMap<>();
-    paramsMerge.put("sourceUrn", edge.getSource().toString());
-    paramsMerge.put("destinationUrn", edge.getDestination().toString());
-    paramsMerge.put("properties", new HashMap<>());
+        "MATCH (source:%s {urn: '%s'}),(destination:%s {urn: '%s'}) MERGE (source)-[r:%s]->(destination) ";
+    String statement =
+        String.format(mergeRelationshipTemplate, sourceType, sourceUrn, destinationType, destinationUrn, edge.getRelationshipType());
+
+    // Add/Update relationship properties
+    String setCreatedOnTemplate;
+    String setcreatedActorTemplate;
+    String setupdatedOnTemplate;
+    String setupdatedActorTemplate;
+    String setPropertyTemplate;
+    final StringJoiner propertiesTemplateJoiner = new StringJoiner(", ");
+    if (edge.getCreatedOn() != null) {
+      setCreatedOnTemplate = "r.createdOn = " + edge.getCreatedOn();
+      propertiesTemplateJoiner.add(setCreatedOnTemplate);
+    }
+    if (edge.getCreatedActor() != null) {
+      setcreatedActorTemplate = "r.createdActor = '" + edge.getCreatedActor() + "'";
+      propertiesTemplateJoiner.add(setcreatedActorTemplate);
+    }
+    if (edge.getUpdatedOn() != null) {
+      setupdatedOnTemplate = "r.updatedOn = " + edge.getUpdatedOn();
+      propertiesTemplateJoiner.add(setupdatedOnTemplate);
+    }
+    if (edge.getUpdatedActor() != null) {
+      setupdatedActorTemplate = "r.updatedActor = '" + edge.getUpdatedActor() + "'";
+      propertiesTemplateJoiner.add(setupdatedActorTemplate);
+    }
+    if (edge.getProperties() != null) {
+      for (Map.Entry<String, Object> entry : edge.getProperties().entrySet()) {
+        if (entry.getValue() instanceof String) {
+          setPropertyTemplate = "r." + entry.getKey() + " = '" + entry.getValue() + "'";
+          propertiesTemplateJoiner.add(setPropertyTemplate);
+        } else {
+          throw new UnsupportedOperationException(String.format(
+              "Tried setting properties on graph edge but property value type is not supported. Key: %s, Value: %s ",
+              entry.getKey(), entry.getValue()));
+        }
+      }
+    }
+    if (!StringUtils.isEmpty(propertiesTemplateJoiner.toString())) {
+      statement = statement + "SET " + propertiesTemplateJoiner;
+    }
```

Comment:
how about we create a new utils file for this class where we can put this logic around `propertiesTemplateJoiner` and checking an `edge` for building a statement? This method is getting pretty hefty